res/controllers/Icon-iControls-scripts.js

```diff
@@ -0,0 +1,187 @@
+                    group: group,
+                    midi: [0xB0+i, 0x0D],
+                    inKey: "rate",
+                    softTakeover: true,
+                });
+                this.knobs[i] = new VelocityPot({
+                    group: group,
+                    midi: [0xB0, 0x10+i],
+                    inKey: "pregain",
+                });
+                this.knobs[i+4] = new VelocityPot({
+                    group: `[QuickEffectRack1_${group}]`,
+                    midi: [0xB0, 0x14+i],
+                    inKey: "super1",
+                });
+                this.pflButton[i] = new components.Button({
+                    group: group,
+                    midi: [0x90, 0x08+i],
+                    key: "pfl",
+                    type: components.Button.prototype.types.toggle,
+                });
+                this.slipButton[i] = new components.Button({
+                    group: group,
+                    midi: [0x90, 0x10+i],
+                    key: "slip_enabled",
+                    type: components.Button.prototype.types.toggle,
+                });
+                this.nextEffectButton[i] = new components.Button({
+                    group: `[QuickEffectRack1_${group}]`,
+                    midi: [0x90, 0x0C+i],
+                    // The manual says this is deprecated, but the replacement
+                    // it gives is for something else, not the quick effects
+                    // rack. Unclear to me what should be done here.
+                    key: "next_chain",
+                });
+                this.superButton[i] = new components.Button({
+                    group: `[QuickEffectRack1_${group}]`,
+                    midi: [0x90, 0x14+i],
+                    key: "enabled",
+                    type: components.Button.prototype.types.toggle,
+                });
+            }
+            this.faders[8] = new components.Pot({
+                group: "[Master]",
+                midi: [0xE8],
+                key: "crossfader",
+                softTakeover: true,
+            });
+
+            // For some reason just this one is a normal 7-bit CC value, so no
+            // need to use the VelocityPot we use on the other knobs on layer 1.
+            this.knobs[8] = new components.Pot({
+                group: "[Master]",
+                midi: [0xB0, 0x0C],
+                key: "gain",
+                softTakeover: true,
+            });
+
+            this.recordButton = new components.Button({
+                group: "[Recording]",
+                midi: [0x90, 0x5F],
+                inKey: "toggle_recording",
+                outKey: "status",
+            });
+        }
+    }
+
+    iControls.init = function() {
+        iControls.controller = new Controller();
+    };
+    iControls.shutdown = function() {
+        iControls.controller.shutdown();
+    };
+})(iControls || (iControls = {}));
+
+// vim:expandtab:shiftwidth=4:tabstop=4
```

Comment:
wouldn't this be better in your own local vim config?
```suggestion
```

Reply:
This format is specifically what the guidelines say Mixxx wants, but it's not what I use for JavaScript in general, so it keeps anyone else who edits with Vim using the same formatting. It's in most of the mappings I've submitted, so I'd prefer to keep it if you're amenable.

Reply:
I'm sure there's some plugin or something you could get for this, but I just use basic vim